flow/connectors/clickhouse/normalize.go

```diff
@@ -167,56 +167,53 @@ func generateCreateTableSQLForNormalizedTable(
 		"`%s` %s, `%s` %s) ENGINE = %s",
 		signColName, signColType, versionColName, versionColType, engine))
 
-	var pkeyStr string
-	pkeys := tableSchema.PrimaryKeyColumns
-	if len(pkeys) > 0 {
-		if len(colNameMap) > 0 {
-			pkeys = slices.Clone(pkeys)
-			for idx, pk := range pkeys {
-				pkeys[idx] = getColName(colNameMap, pk)
-			}
-		}
-		pkeyStr = strings.Join(pkeys, ",")
+	orderByColumns := getOrderedOrderByColumns(tableMapping, tableSchema.PrimaryKeyColumns)
+
+	if len(orderByColumns) > 0 {
+		orderByStr := strings.Join(orderByColumns, ",")
 
 		stmtBuilder.WriteString("PRIMARY KEY (")
-		stmtBuilder.WriteString(pkeyStr)
+		stmtBuilder.WriteString(orderByStr)
+		stmtBuilder.WriteString(") ")
+
+		stmtBuilder.WriteString("ORDER BY (")
+		stmtBuilder.WriteString(orderByStr)
 		stmtBuilder.WriteString(") ")
 	}
 
-	var orderby []*protos.ColumnSetting
+	return stmtBuilder.String(), nil
+}
+
+// Returns a list of order by columns ordered by their ordering, and puts the pkeys at the end.
```

Comment:
shouldn't this be the opposite?

``` DB::Exception: Primary key must be a prefix of the sorting key```

Reply:
we are making pkey and order key same.

Reply:
Oh ok, is there any reason for this change?

Reply:
Also,

```If we would have specified only the sorting key, then the primary key would be implicitly defined to be equal to the sorting key.`

can drop the clause entirely then?